src/SFtp.cc

```diff
@@ -1191,8 +1194,6 @@ int SFtp::Read(Buffer *buf,int size)
       int bytes_allowed=rate_limit->BytesAllowedToGet();
       if(size1>bytes_allowed)
 	 size1=bytes_allowed;
-      if(size1==0)
```

Comment:
The patch wraps the code in `if(file_buf && file_buf->Size()>0)` and size1 is file_buf->Size(), so this looked to me like a redundant code.